tls/handshake_client.go

```diff
@@ -515,6 +515,10 @@ func (c *Conn) clientHandshake() error {
 		if err := hs.doFullHandshake(); err != nil {
 			return err
 		}
+		if c.config.CertsOnly {
+			// All done
+			return nil
```

Comment:
This should return a package-level error that clients can compare against, similar to `io.EOF`. Something like `ErrCertsOnly`, rather than `nil`.

Reply:
@dadrian that makes sense to me.  Thanks for the feedback, I'll see if I can generate a new commit with that change.

Reply:
Just to make sure I'm on the same page, I think the idea is that the short circuit in line 592 should have something like

```
return errors.New("tls: handshake abandoned per TLSCertsOnly option: " + certsOnlyErr.Error())
```

And then the lines in 518 -- 520 would just be removed (because 515,516 already passes along errors from hs.doFullHandshake().

Is that what you're thinking?